dynamiqs/sparse.py

```diff
@@ -122,3 +127,68 @@ def _check_compatible_dims(dims1: tuple[int, ...], dims2: tuple[int, ...]):
         raise ValueError(
             f'QArrays have incompatible dimensions. Got {dims1} and {dims2}.'
         )
+
+
+def to_dense(sparse: SparseQArray) -> Array:
+    r"""Returns the input matrix in the Dense format.
+
+    Parameters:
+        sparse: A sparse matrix, containing diagonals and their offsets.
+
+    Returns:
+        Array: A dense matrix representation of the input sparse matrix.
+    """
+    N = sparse.dims[0]
+    out = jnp.zeros((N, N))
+    for offset, diag in zip(sparse.offsets, sparse.diags):
+        start = max(0, offset)
+        end = min(N, N + offset)
+        out += jnp.diag(diag[start:end], k=offset)
+    return out
+
+
+def to_sparse(other: ArrayLike) -> SparseQArray:
+    r"""Returns the input matrix in the SparseDIA format.
+
+    This should be used when a user wants to turn a dense matrix that
+    presents sparse features in the SparseDIA custom format.
+
+    Args:
+        other: NxN matrix to turn from dense to SparseDIA format.
+
+    Returns:
+        SparseDIA object which has 2 main attributes:
+            object.diags: Array where each row is a diagonal.
+            object.offsets: tuple of integers that represents the
+                            respective offsets of the diagonals
+    """
+    diagonals = []
+    offsets = []
+
+    N = other.shape[0]
+    offset_range = 2 * N - 1
+    offset_center = offset_range // 2
+
+    if not isinstance(other, Array):
+        other = jnp.asarray(other.array)
```

Comment:
```suggestion
    other = jnp.asarray(other.array)
```
(should be at the beginning of the function too).